src/main/java/net/minecraftforge/server/ServerLifecycleHooks.java

```diff
@@ -113,6 +113,10 @@ public static void expectServerStopped()
 
     public static void handleServerStopped(final MinecraftServer server)
     {
+        if (StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).getCallerClass() != MinecraftServer.class)
+        {
+            throw new IllegalCallerException("This is an internal forge method and cannot be used by mods");
```

Comment:
Although the ServerStoppedEvent is just below here in the code, it might still be a good idea to mention it in the exception in case they don’t look at this method’s contents.